### PR TITLE
Handle complex quoting in extra args in pull cli

### LINKED ISSUE #40729: ansible-pull  extra-vars mistakes JSON for YAML
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

ansible-pull

##### ANSIBLE VERSION

```
ansible-pull 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-pull
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

##### OS / ENVIRONMENT

Ubuntu 18.04 server

##### STEPS TO REPRODUCE

```yaml
export ANSIBLE_EXTRA_VARS='{"docker_registries_login": [{ "docker_password": "'"${PASSWORD}"'", "docker_username": "'"${USER}"'", "docker_registry_url":"repository-manager.company.com:5001"}], "docker_registries_logout": [{ "docker_password": "'"${PASSWORD}"'", "docker_username": "'"${USER}"'", "docker_registry_url":"repository-manager.company.com:5001"}] }'

ansible-pull --accept-host-key                            \
             --checkout "${ANSIBLE_REPOSITORY_BRANCH}"    \
             --clean                                      \
             --extra-vars "${ANSIBLE_EXTRA_VARS}" \
             --full                                       \
             --purge                                      \
             --url "${PACKER_ANSIBLE_REPOSITORY_URL}"     \
             -vvv                                         \
             "${PACKER_ANSIBLE_PULL_PLAYBOOK}"
```

##### EXPECTED RESULTS

ansible-pull should have taken the Ansible extra vars passed as JSON and then run normally. What happens is that it fails because it tries parsing the JSON as YAML.

Oddly enough though, passing Ansible extra vars in the same manner with ansible-playbook works fine for me. The problem seems to be in the docker_registry_url value having a colon in it.

##### ACTUAL RESULTS

```
ansible-pull 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-pull
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
Using /etc/ansible/ansible.cfg as config file
Starting Ansible Pull at 2018-05-25 18:06:49
/usr/bin/ansible-pull --accept-host-key --checkout 0.2.0 --clean --extra-vars {"docker_registries_login": [{ "docker_password": "<pw>", "docker_username": "user", "docker_registry_url":"repository-manager.company.com:5001"}], "docker_registries_logout": [{ "docker_password": "<pw>", "docker_username": "user", "docker_registry_url":"repository-manager.company.com:5001"}] } --full --purge --url ssh://git@bitbucket.org/myuser/myrepo.git -vvv local.yml
ansible 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
Using /etc/ansible/ansible.cfg as config file
Parsed localhost, inventory source with host_list plugin
ERROR! Syntax Error while loading YAML.
  found unexpected ':'

The error appears to have been in '<string>': line 1, column 197, but may
be elsewhere in the file depending on the exact syntax problem.
```
I resolved the issue by dumping the JSON to file and pointing --extra-vars at the file but I am opening this as a bug either way because the same JSON string works with ansible-playbook's --extra-vars when passed directly to ansible-playbook in the same manner.

Maybe ansible-pull isnt properly passing on/escaping the value to ansible-playbook?

### PR DESCRIPTION
**Pull request by bcoca**
fixes #40729

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

bin/ansible-pull

### CODE REVIEW

File: test/integration/targets/pull/runme.sh [hunk 1]
```diff
@@ -23,21 +23,33 @@ cd "${repo_dir}"
...
-# make sure one host was run
-if ! egrep 'localhost.*ok' "${temp_log}"; then
-    echo "Did not find host localhost in output."
-    exit 1
-fi
+pass_tests
+
+# ensure complex extra vars work
+PASSWORD='test'
+JSON_EXTRA_ARGS='{"docker_registries_login": [{ "docker_password": "'"${PASSWORD}"'", "docker_username": "'"${USER}"'", "docker_registry_url":"repository-manager.company.com:5001"}], "docker_registries_logout": [{ "docker_password": "'"${PASSWORD}"'", "docker_username": "'"${USER}"'", "docker_registry_url":"repository-manager.company.com:5001"}] }'
```
Comment (mattclay) [inline test/integration/targets/pull/runme.sh:52]: CI failure in integration tests due to:

> 11:18 ./runme.sh: line 51: USER: unbound variable

Comment (bcoca) [reply test/integration/targets/pull/runme.sh:52]: @mattclay if USER is not being set, i consider this a bug in the environment

Comment (bcoca) [reply test/integration/targets/pull/runme.sh:52]: we actually rely on it in our connection code

Comment (mattclay) [reply test/integration/targets/pull/runme.sh:52]: Perhaps we shouldn't rely on the `USER` variable being set. It's not available in all environments. For example, it doesn't seem to be present in Docker by default:

```bash
$ docker run -it centos:7 env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=d817fab2e123
TERM=xterm
HOME=/root
```

Similar results (without `USER`) occur for `fedora:29` and `ubuntu:16.04` containers, as well as when running `/bin/bash` interactively first in those containers.

Comment (bcoca) [reply test/integration/targets/pull/runme.sh:52]: i'll update test, but that still seems like there is something wrong with those environments

**Review by jimi-c — APPROVED** [review]

**Pull request merged**